Image correctly being sent from History tab to img2img

anastasiuspernat opened the pull request:
A proposal to fix issues https://github.com/AUTOMATIC1111/stable-diffusion-webui/issues/2941, https://github.com/AUTOMATIC1111/stable-diffusion-webui/issues/2892, https://github.com/AUTOMATIC1111/stable-diffusion-webui/issues/2783, https://github.com/AUTOMATIC1111/stable-diffusion-webui/issues/3272

It correctly sends images from History and PNG Info tabs to img2img (for inpainting, outpainting etc). Right now it's not working, only parameters are transferred. So img2img does not work after clicking "send to img2img".

anastasiuspernat committed:
Fixed image not appearing when sending from History & PNG Info to img2img

AUTOMATIC1111 commented:
PNG info is not supposed to sen the image, just the params.

Also did you just take a function from another file and copy-paste it into images_history?

anastasiuspernat commented:
> PNG info is not supposed to sen the image, just the params.
> 
> Also did you just take a function from another file and copy-paste it into images_history?

Thanks for reviewing!

1) I see! How about I load an image (to PNG Info) and want to outpaint it in img2img?
2) Yes I did :-( It was hard to understand what's going on in images_history. But I'll remove the copy

AUTOMATIC1111 commented:
> I see! How about I load an image with its metadata, keywords etc (to PNG Info) and want to outpaint it in img2img, with the same metadata?

You drag the image into the img2img interface.

anastasiuspernat commented:
> > I see! How about I load an image with its metadata, keywords etc (to PNG Info) and want to outpaint it in img2img, with the same metadata?
> 
> You drag the image into the img2img interface.

Yeah! But if it's on another computer (and webui accessed thru network)? Also is it easier to click one button in PNG Info in any case?

Maybe create another PR?

AUTOMATIC1111 commented:
if the image is on another computer then you didn't drag it into PNG info.

anastasiuspernat committed:
Removed PNG Info img transfer, removed duplicate function

anastasiuspernat commented:
> if the image is on another computer then you didn't drag it into PNG info.

Yeah I see! Makes sense. I removed PNG Info image transfer and also removed duplicate code.

anastasiuspernat committed:
Removed unnecessary imports

jyapayne commented:
@anastasiuspernat I would like this feature to work as well. Any chance you could port it to the new image browser?

anastasiuspernat commented:
> @anastasiuspernat I would like this feature to work as well. Any chance you could port it to the new image browser?

Image Browser now a separate extension that doesn't have access to image data. As I understand @yfszzx proposed another change here https://github.com/AUTOMATIC1111/stable-diffusion-webui/pull/3768 that will fix this issue. And then their extension would need another update.

anastasiuspernat commented:
I'm closing PR as it's now fixed in the Image Browser extension.
